atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -97,122 +102,65 @@ def run(self):
-                "incomplete and may cause race conditions during build", image
-            )
-            return
-
-        manifest_list_dict = manifest_list.json()
-
-        try:
-            arch_to_platform = get_goarch_to_platform_mapping(self.workflow)
-        except KeyError:
-            self.log.warning('Cannot collect platforms digests for parent images '
-                             'because platform descriptors are not defined')
-            return
-
-        for manifest in manifest_list_dict['manifests']:
-            arch = manifest['platform']['architecture']
+        if manifest_list:
+            manifest_digest = get_checksums(BytesIO(manifest_list.content), ['sha256'])
+            media_type = get_manifest_media_type('v2_list')
+        else:
+            digests_dict, _ = get_all_manifests(image, image.registry,
+                                                self.parent_registry_insecure,
+                                                self.parent_registry_dockercfg_path,
+                                                versions=('v2'))
+            media_type = get_manifest_media_type('v2')
             try:
-                present_platform = arch_to_platform[arch]
+                manifest_digest = digests_dict['v2']
             except KeyError:
-                self.log.warning("Cannot map architecture='{}' to platform".format(arch))
-                continue
-            else:
-                digest = manifest['digest']
-                self.log.info("Collecting digest for image '%s' ('%s'): '%s'",
-                              image, present_platform, digest)
-                parents_digests.update_image_digest(
-                    image, present_platform, digest
-                )
-
-    def _get_image_for_different_arch(self, image, platform):
-        """Get image from random arch
-
-        This is a workaround for aarch64 platform, because orchestrator cannot
-        get this arch from manifests lists so we have to provide digest of a
-        random platform to get image metadata for orchestrator.
-
-        For standard platforms like x86_64, ppc64le, ... this method returns
-        the corresponding digest
-
-        """
-        parents_digests = self.workflow.builder.parent_images_digests
-        try:
-            digests = parents_digests.get_image_digests(image)
-        except KeyError:
-            return None
+                raise RuntimeError('Could not extract manifest list or '
+                                   'v2 schema 2 digest for {}'.format(image_str))
 
-        if not digests:
-            return None
-        platform_digest = digests.get(platform)
-        if platform_digest is None:
-            # exact match is not found, get random platform
-            platform_digest = tuple(digests.values())[0]
-
-        new_image = ImageName.parse(platform_digest)
-        return new_image
+        parent_metadata = {'digests': manifest_digest, 'media_type': media_type}
```

Comment:
May happen a case when `v2_list` digest will be overwritten by `v2` manifest digest

Reply:
Maybe we should error here if value already exists and we are trying to replace it with something else

Reply:
mmmm how it could be overwritten?
it either has manifest_list or it doesn't

Reply:
accidents happen, server may return different type on second call. Or optimization can be done here, if record exists, don't call registry again.

Reply:
for multistage builds

Reply:
parent_images contains unique images btw

Reply:
ok, then

Reply:
I changed the structure here.

* re-use the parent_images_digests param to avoid having osbs-client changes (it is also a better name)
* each parent image dict will have the media_type as a key and digest as value.

Thoughts? @MartinBasti @rcerven

Reply:
Sounds good

Reply:
done